File: api/Models/ProfessorSubject.cs
using System.ComponentModel.DataAnnotations.Schema;
namespace api.Models
{
    [Table("Profesor_predmet")]
    public class ProfessorSubject
    {
        [Column("id_profesor_predmet")]
        public int Id { get; set; }
        [Column("id_korisnika")]
        public int userId { get; set; }
        [Column("id_predmeta")]
        public int subjectId { get; set; }
    }
}

File: api/app/Models/ProfessorSubject.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ProfessorSubject extends Model
{
    use HasFactory;
    const TABLE = 'professor_subjects';
    protected $table = self::TABLE; 
    protected $fillable = [
        'user_id',
        'subject_id',
    ];
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function subject(): BelongsTo
    {
        return $this->belongsTo(Subject::class);
    }
}